tests/test_iot/test_iot.py

```diff
@@ -2210,3 +2213,40 @@ def test_topic_rule_delete(self):
 
         res = client.list_topic_rules()
         res.should.have.key("rules").which.should.have.length_of(0)
+
+    @mock_iot
+    def test_deprecate_undeprecate_thing_type(self):
+        client = boto3.client("iot", region_name="ap-northeast-1")
+        thing_type_name = "my-type-name"
+        client.create_thing_type(
+            thingTypeName=thing_type_name,
+            thingTypeProperties={"searchableAttributes": ["s1", "s2", "s3"]},
+        )
+
+        res = client.describe_thing_type(thingTypeName=thing_type_name)
+        res["thingTypeMetadata"]["deprecated"].should.equal(False)
+        client.deprecate_thing_type(
+            thingTypeName=thing_type_name,
+            undoDeprecate=False,
+        )
+
+        res = client.describe_thing_type(thingTypeName=thing_type_name)
+        res["thingTypeMetadata"]["deprecated"].should.equal(True)
+
+        client.deprecate_thing_type(
+            thingTypeName=thing_type_name,
+            undoDeprecate=True,
+        )
+
+        res = client.describe_thing_type(thingTypeName=thing_type_name)
+        res["thingTypeMetadata"]["deprecated"].should.equal(False)
+
+    @mock_iot
+    def test_deprecate_thing_type_not_exist(self):
+        client = boto3.client("iot", region_name="ap-northeast-1")
+        thing_type_name = "my-type-name"
+        with pytest.raises(client.exceptions.ResourceNotFoundException):
```

Comment:
We should catch `botocore.exceptions.ClientError` here, as that's the error class that boto3 uses.
Ideally we also assert that the right error is thrown:
```
with pytest.raises(ClientError) as ex:
    ....
err = ex.value.response["Error"]
err["Message"].should.equal("...")
```

Reply:
Hi, thanks for that, indeed i have a liner issue - if the next one fails i'll search for the correct one.
the ClientError is the uglier way to catch boto3 exceptions, some clients do offer the "specific" exception:

https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#parsing-error-responses-and-catching-exceptions-from-aws-services

as shown here 
![image](https://user-images.githubusercontent.com/33662254/129762175-7cac426f-afd3-4284-af01-dc019d9afff3.png)

Reply:
Fair enough - I misread, and thought it referred to our custom `moto.iot.exceptions.ResourceNotFoundException`